spec/requests/rates_spec.rb

```diff
@@ -0,0 +1,240 @@
+      tiers.zip(parameter_tiers).each do |tier, tier_parameters_original|
+        tier_parameters = tier_parameters_original.dup
+
+        tier_attributes = tier.attributes.except('id')
+        tier_parameters['chargeback_rate_detail_id'] = response.parsed_body["results"][0]['id'].to_i
+        tier_parameters['finish'] = Float::INFINITY if tier_parameters['finish'] == 'infinity'
+
+        expect(tier_attributes).to eq(tier_parameters)
+      end
+
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  let(:currency_2) { FactoryBot.create(:currency) }
+
+  it "can edit a chargeback rate detail through POST" do
+    chargeback_rate = FactoryBot.create(:chargeback_rate)
+    chargeback_rate_detail = FactoryBot.build(:chargeback_rate_detail, :description => "rate_0", :chargeable_field => field)
+    chargeback_tier = FactoryBot.create(:chargeback_tier, :chargeback_rate_detail_id => chargeback_rate_detail.id,
+                                        :start => 0, :finish => Float::INFINITY, :fixed_rate => 0.0,
+                                        :variable_rate => 0.0)
+    chargeback_rate_detail.chargeback_tiers = [chargeback_tier]
+    chargeback_rate_detail.save
+
+    parameters_relations = [
+      {:detail_currency  => {:id=> currency_2.id},
+       :chargeback_tiers => parameter_tiers,
+       :chargeable_field => {:id => chargeable_field.id},
+       :chargeback_rate  => {:id => chargeback_rate.id}},
+      {:detail_currency  => {:href => api_currency_url(nil, currency_2)},
+       :chargeback_tiers => parameter_tiers,
+       :chargeable_field => {:href => api_chargeable_field_url(nil, chargeable_field)},
+       :chargeback_rate  => {:href => api_chargeback_url(nil, chargeback_rate)}}
+    ]
+
+    parameters_relations.each do |relations|
+      api_basic_authorize action_identifier(:rates, :edit)
+      relations[:description] = "rate_1"
+      post api_rate_url(nil, chargeback_rate_detail), :params => gen_request(:edit, relations)
+
+      expect(response.parsed_body["description"]).to eq("rate_1")
+      expect(response.parsed_body["chargeback_rate_detail_currency_id"].to_i).to eq(currency_2.id)
+      expect(response.parsed_body['chargeable_field_id'].to_i).to eq(chargeable_field.id)
+      expect(response.parsed_body['chargeback_rate_id'].to_i).to eq(chargeback_rate.id)
+
+      expect(response).to have_http_status(:ok)
+      expect(chargeback_rate_detail.reload.description).to eq("rate_1")
+
+      tiers = chargeback_rate_detail.reload.chargeback_tiers.sort_by(&:start)
+      tiers.zip(parameter_tiers).each do |tier, tier_parameters_original|
+        tier_parameters = tier_parameters_original.dup
+
+        tier_attributes = tier.attributes.except('id')
+        tier_parameters['chargeback_rate_detail_id'] = response.parsed_body['id'].to_i
+        tier_parameters['finish'] = Float::INFINITY if tier_parameters['finish'] == 'infinity'
```

Comment:
just curious, I see this pattern from before, can this logic be put in a method ?